options.js

```diff
@@ -1,139 +1,193 @@
+				var textboxhl = document.optform.textboxhl.checked;
+				var onlyreading = document.optform.onlyreading.checked;
+				var minihelp = document.optform.minihelp.checked;
+				var disablekeys = document.optform.disablekeys.checked;
+				var kanjicomponents = document.optform.kanjicomponents.checked;
+				var lineEnding = document.optform.lineEnding.value;
+				var copySeparator = document.optform.copySeparator.value;
+				var maxClipCopyEntries = document.optform.maxClipCopyEntries.value;
+				var popupDelay = getPopUpDelay();
+				var popupLocation =document.optform.popupLocation.value;
+				var showOnKey = document.optform.showOnKey.value;
+				var kanjiInfoObject = {};
+				var kanjiInfoLabelList = chrome.extension.getBackgroundPage().rcxDict.prototype.kanjiInfoLabelList;
+				var kanjiinfoarray = new Array(kanjiInfoLabelList.length/2);
+
+				// Setting Kanji values
+				for (i = 0; i*2 < kanjiInfoLabelList.length; i++) {
+					kanjiInfoObject[kanjiInfoLabelList[i * 2]] = document.getElementById(kanjiInfoLabelList[i * 2]).checked;
+					kanjiinfoarray[i] = kanjiInfoObject[kanjiInfoLabelList[i * 2]];
+				}
+
+				chrome.storage.sync.set({
+					// Saving General options
+					"popupcolor": popupcolor,
+					"highlight": highlight,
+					"textboxhl": textboxhl,
+					"onlyreading": onlyreading,
+					"minihelp": minihelp,
+					"disablekeys": disablekeys,
+					"kanjicomponents": kanjicomponents,
+					"kanjiInfo":kanjiInfoObject,
+					"popupLocation":popupLocation,
+					// Saving Copy to Clipboard settings
+					"lineEnding": lineEnding,
+					"copySeparator": copySeparator,
+					"maxClipCopyEntries": maxClipCopyEntries,
+					"popupDelay": popupDelay,
+					"showOnKey":showOnKey
+				});
+
+				chrome.extension.getBackgroundPage().rcxMain.config.css = popupcolor;
+				chrome.extension.getBackgroundPage().rcxMain.config.highlight = highlight;
+				chrome.extension.getBackgroundPage().rcxMain.config.textboxhl = textboxhl;
+				chrome.extension.getBackgroundPage().rcxMain.config.onlyreading = onlyreading;
+				chrome.extension.getBackgroundPage().rcxMain.config.minihelp = minihelp;
+				chrome.extension.getBackgroundPage().rcxMain.config.popupDelay = popupDelay;
+				chrome.extension.getBackgroundPage().rcxMain.config.disablekeys = disablekeys;
+				chrome.extension.getBackgroundPage().rcxMain.config.showOnKey = showOnKey;
+				chrome.extension.getBackgroundPage().rcxMain.config.kanjicomponents = kanjicomponents;
+				chrome.extension.getBackgroundPage().rcxMain.config.kanjiinfo = kanjiinfoarray;
+				chrome.extension.getBackgroundPage().rcxMain.config.lineEnding = lineEnding;
+				chrome.extension.getBackgroundPage().rcxMain.config.copySeparator = copySeparator;
+				chrome.extension.getBackgroundPage().rcxMain.config.maxClipCopyEntries = maxClipCopyEntries;
+				chrome.extension.getBackgroundPage().rcxMain.config.popupLocation = popupLocation;
+
+			}
+
+
+			function getPopUpDelay(){
+				// Todo: Why do we need a popup delay?
```

Comment:
It's so if you're just moving your mouse from one place to another, you don't get a ton of japanese pop ups, but if you pause your mouse over a word then the you will see the popup.

Reply:
Copy that!